oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -479,7 +479,10 @@ private static QueryBuilder tokenToQuery(String text, String fieldName, PlanResu
             // TODO: do we need this if all the analyzed fields are queried?
             ret = in.should(matchQuery(fieldName, text));
```

Comment:
shouldn't we have an AND operator here too ?

Reply:
correct, let me add a test to capture this case